great_expectations/zep/interfaces.py

```diff
@@ -14,10 +15,12 @@ def __init__(self, name: str) -> None:
         self.name = name
 
 
-@runtime_checkable
-class Datasource(Protocol):
-    execution_engine: ExecutionEngine
-    asset_types: List[Type[DataAsset]]
+class Datasource(metaclass=MetaDatasource):
+    # class attrs
+    execution_engine: ClassVar[ExecutionEngine]
```

Comment:
Let's keep `execution_engine` an instance variable. Each instance should instantiate its own in general.